src/app/containers/Timestamp/relativeTimestamp.test.js

```diff
@@ -0,0 +1,86 @@
+import relativeTimestamp from './relativeTimestamp';
+
+const timeStampGenerator = timedifference =>
+  Date.now() - timedifference.milliseconds * timedifference.magnitude;
+
+const timestampWithDiffInHours = hours =>
+  timeStampGenerator({
+    milliseconds: hours,
+    magnitude: 60 * 60 * 1000,
+  });
+
+const timestampWithDiffInMins = mins =>
+  timeStampGenerator({
+    milliseconds: mins,
+    magnitude: 60 * 1000,
+  });
+
+const relativeBehaviour = (description, input, expectedOutput) => {
+  it(description, () => {
+    const result = relativeTimestamp(input);
+    expect(result).toEqual(expectedOutput);
+  });
+};
+
+describe('relativeTimestamp', () => {
+  it('returns a string which ends in ago', () => {
+    const timestamp = timestampWithDiffInMins(1);
+    const result = relativeTimestamp(timestamp);
+    expect(typeof result).toEqual('string');
+    expect(result.split(' ').pop()).toEqual('ago');
+  });
+
+  relativeBehaviour(
+    'returns 1 minute ago',
+    timestampWithDiffInMins(1),
+    '1 minute ago',
+  );
+
+  relativeBehaviour(
+    'returns 5 minutes ago',
+    timestampWithDiffInMins(5),
+    '5 minutes ago',
+  );
+
+  relativeBehaviour(
+    'returns 1 hour ago',
+    timestampWithDiffInHours(1),
+    '1 hour ago',
+  );
+
+  relativeBehaviour(
+    'returns 5 hours ago',
+    timestampWithDiffInHours(5),
+    '5 hours ago',
+  );
+
+  relativeBehaviour(
+    'returns 5 hours ago',
+    timestampWithDiffInHours(5),
+    '5 hours ago',
+  );
+
+  relativeBehaviour(
+    'returns 1 minute ago for 10 seconds',
+    timeStampGenerator({
+      milliseconds: 10,
+      magnitude: 1000,
```

Comment:
I think this way of generating a timestamp is fiddly to read - better to just pass what you want:

```js
timestampGenerator({ seconds: 10 });
timestampGenerator({ mins: 3 });
timestampGenerator({ hours: 3, mins: 32, seconds: 41 });
// etc
```

...rather than always passing milliseconds but sometimes passing a magnitude

Reply:
Improved this and caught an edge case 👍